### PR TITLE
Update LinearGradientBrush based on FlowDirection 

### LINKED ISSUE #11603: LinearGradientBrush Doesn't Respect Right To Left FlowDirection
### Description
**LinearGradientBrush** Doesn't Respect Right To Left **FlowDirection** . when **FlowDirection** is **Right To Left** , Gradient Brush isn't drawn from Right To Left . _in fact , Gradient Brush Direction is always from Left To Right despite the **FlowDirection** ._
### Steps to Reproduce

1. Define a Frame and set it's Background to **LinearGradientBrush** (Horizontally):
```
<Frame HeightRequest="120" Margin="5">
            <Frame.Background>
                    <LinearGradientBrush StartPoint="0,0" EndPoint="1,0">
                        <GradientStop Color="Red" Offset="0.0"/>
                        <GradientStop Color="Blue" Offset="1.0"/>
                    </LinearGradientBrush>
            </Frame.Background>
</Frame>
```
2. Change **FlowDirection** to **_RightToLeft_**
3. LinearGradientBrush  direction is still from Left To Right . the Views inside Frame are laid out from Right To Left.

### Expected Behavior
When **FlowDirection** is **RightToLeft** , **LinearGradientBrush**  must be drawn from _Right To Left_ .(Coordinate center must be TOP-RIGHT and all calculations must be relative toward this point)
### Actual Behavior
When **FlowDirection** is **_RightToLeft_** , LinearGradientBrush is still drawn **from Left To Right**. it is noticeable that child views inside Frame is laid out from right to left , but GradientBrush itself doesn't behave so.
### Basic Information

- Version with issue: Xamarin.Forms 4.8.0.1187-pre2
- Last known good version:
- IDE: Visual Studio 2019 16.7 Prev 2.0
- Platform Target Frameworks: 
  - OS: 13,5
  - Android: 9.0 
- Android Support Library Version: 
- Nuget Packages: Xamarin.Forms 4.8.0.1187-pre2
- Affected Devices:

### Screenshots

**ANDROID:**
Android Gradient RTL_LTR

**iOS**
iPAD_Gradient_RTL _LTR

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Update LinearGradientBrush based on FlowDirection .

_Work in progress_

### Issues Resolved ### 

- fixes #11603 

### API Changes ###
 
 None

### Platforms Affected ### 

- Core/XAML (all platforms)
- iOS
- macOS
- Android
- UWP

### Behavioral/Visual Changes ###

FlowDirection defines not only how text flows in a textual element but also the flow direction of almost every other UI element. With this changes, changing the FlowDirection also affects to a LinearGradientBrush.

### Before/After Screenshots ### 

brush-flowdirection-ios

flow-brush-macos

### Testing Procedure ###
Launch Core Gallery and navigate to the brushes gallery. Select the "" and apply FlowDirection changes to see the behavior.

### PR Checklist ###
- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Platform.Android/Renderers/CircularProgress.cs [hunk 1]
```diff
@@ -53,14 +53,17 @@ public void SetColor(Color color)
 			}
 		}
 
-		public void SetBackground(Color color, Brush brush)
+		public void SetBackground(Color color, Brush brush, FlowDirection flowDirection)
```
Comment (samhouts) [inline Xamarin.Forms.Platform.Android/Renderers/CircularProgress.cs:56]: breaking change

**Review by samhouts — CHANGES_REQUESTED** [review]
Can you make sure that any public signature changes are done with overloads instead of changing them?

Also, if we're changing API, please target 5.0.0.

Thanks!

**Comment by jsuarezruiz** [comment]
Retarget to 5.0.0.

**Comment by rmarinho** [comment]
Build is failing @jsuarezruiz

**Comment by jsuarezruiz** [comment]
@rmarinho Fixed the build error.

**Comment by rmarinho** [comment]
need to resolve conflicts now @jsuarezruiz  :)

**Comment by samhouts** [comment]
o no....more conflicts! 😱

**Comment by jfversluis** [comment]
@jsuarezruiz there seems to be a lot of changes here. Should we skip this for Forms and move it to .NET MAUI?

**Comment by jfversluis** [comment]
This is too big of a change to still add to Xamarin.Forms at this point in time. Thanks all for the time and effort here!

**Pull request closed without merging**